salesforce/backend/query.py

```diff
@@ -483,6 +513,12 @@ def execute(self, q, args=()):
             self.results = iter([])
 
     def execute_select(self, q, args):
+        if q == "SELECT django_migrations.app, django_migrations.name FROM django_migrations":
```

Comment:
You are right about documentation. Any help will be appreciated, even to track the project only and to review and early or constructive comments. My language skills are problematic when I described a complex "moving target" and other people can not understand it in README clearly. They respond too late or propose a fix by removing. Not too motivating. I started then to write more info to Wiki and even more info to [hynekcer sf wiki](https://github.com/hynekcer/django-salesforce/wiki/), where all original project topics are linked together and some independent topics (urls) are added. (My repository looks messy but the Wiki is a text for other people or a concept of this) I can write code or documentation, but usually it is very innefficient to try to do it synchronized. The objective is to write a pseudo-documentation during development, use it for discussion and later to be rewritten to documentation by other person. I have no ambition to start multi-page documentation, even that it would be useful to differentiate e.g. easy install :-) from the reference.
 
It even would not help me to have a documentation for writing custom db backend, because nobody reasonable from Django team would promise any interface for NoSQL databases. In the recent Django versions since 1.8 you can see a backend specific code not only in `django/db/backends/', but also in 'django/db/models/', e.g. many `as_mysql` methods. There is no clean way to write an additional backend without tricky code. I remember previous versions of MongoDB for Django, where it was even more tricky due to Mongo driver for patched Django, up to version 1.5.

"Tricky"? As little as possible. Example: Your code is by the standard recommended style, e.g. the try/except branch for new/old code. My experience is, when I must observe undocumented information and also must regularly remove obsoleted code after previous Django versions, that the explicit style ("if django version > something; else;) is better for tests, debugging and also to get more useful tracebacks in issues.

Migrations: First: Search the string "migrat" in README (4 lines in 3 paragraphs) and marginally also [Introspection and Special Attributes...](https://github.com/django-salesforce/django-salesforce/wiki/Introspection-and-Special-Attributes-of-Fields) (1 paragraph) ... (after you read) ...
Migrations are more and more important in Django, even the current "fake migrations" for Salesforce, if you for some reason want to use SalesforceModel with another database, e.g fast in memory tests, data copying etc. (see issues 56, 97, 98. 111, 167 that it is important.) This has been supported for a long time until Django 1.10 and fixed just now. My personal style to frequently run inspectdb on the current SFDC and run a filter to select important/unimportant fields for the final model. If SFDC should alternate with another db in development or maintenance tasks, then run makemigrations an migrate to get the same model in that database. If you can try it and explain it human way, do it please. :-)